src/profile-logic/process-profile.js

```diff
@@ -507,9 +508,34 @@ function _processFrameTable(
 /**
  * Explicitly recreate the stack table here to help enforce our assumptions about types.
  */
-function _processStackTable(geckoStackTable: GeckoStackStruct): StackTable {
+function _processStackTable(
+  geckoStackTable: GeckoStackStruct,
+  frameTable: FrameTable,
+  categories: CategoryList
+): StackTable {
+  // Compute a non-null category for every stack
+  const greyCategory = categories.findIndex(c => c.color === 'grey') || 0;
+  const categoryColumn = [];
```

Comment:
This way of creating arrays does not pre-allocate space, it only sets `length` to a different value. I've found [a benchmark](https://jsperf.com/array-growth-dynamic-vs-preset) that compares the two; both approaches have the same performance and grow the buffer dynamically.

Reply:
so I actually got a discussion with @nbp about this. If you set the length, it will preallocate, but with a "max" boundary (around 1200 slots maximum IIRC).
I don't know what the behavior is when resizing: is it resizing with bigger chunks when it then resizes ? Maybe @nbp can help answering this question.

Also the benchmark is slightly different in that you're using `push` in this code. This may trigger different behavior.

That said we can also look at this again later if we find there's a perf issue.

Reply:
Indeed using `new Array(nb)` will reserve space up to `nb` for small numbers.  Using `push` will amortize the allocation cost by doubling the size.